utils/crypto/ledger/ledger.go

```diff
@@ -40,21 +42,40 @@ func addressPath(index uint32) string {
 }
 
 func (l *Ledger) Address(hrp string, addressIndex uint32) (ids.ShortID, error) {
-	_, hash, err := l.device.GetPubKey(addressPath(addressIndex), true, hrp, "")
+	resp, err := l.device.GetPubKey(addressPath(addressIndex), true, hrp, "")
 	if err != nil {
 		return ids.ShortEmpty, err
 	}
-	return ids.ToShortID(hash)
+	return ids.ToShortID(resp.Hash)
 }
 
 func (l *Ledger) Addresses(addressIndices []uint32) ([]ids.ShortID, error) {
+	if l.epk == nil {
+		pk, chainCode, err := l.device.GetExtPubKey(rootPath, false, "", "")
+		if err != nil {
+			return nil, err
+		}
+		l.epk = &bip32.Key{
```

Comment:
We don't need any of the other fields (version, depth, etc)?

Reply:
Yup. Not needed at all.